### ISSUE TITLE
SECURITY: bad regex pattern in 'sklearn/externals/_arff.py' will cause 'ReDos' security problem.

### ISSUE THREAD

**Issue by jnothman** [renatopp/liac-arff#117]
Posted at https://github.com/scikit-learn/scikit-learn/issues/19522 by @leveryd, `_RE_TYPE_NOMINAL` presents a ReDos risk.

**Comment by ogrisel** [comment]
```
_RE_TYPE_NOMINAL = re.compile(r'^\{\s*((\".*\"|\'.*\'|\S*)\s*,\s*)*(\".*\"|\'.*\'|\S*)\s*\}$', re.UNICODE)
```

at line: https://github.com/renatopp/liac-arff/blob/35816f5c393a6a04ed6fcc2d41b0309433fa4cd3/arff.py#L166

**Comment by mfeurer** [comment]
Fixed via #118. Will be available in the next release.
